crates/abq_cli/src/main.rs

```diff
@@ -332,23 +372,42 @@ async fn abq_main() -> anyhow::Result<ExitCode> {
             let tls_cert = read_opt_path_bytes(tls_cert)?;
             let tls_key = read_opt_path_bytes(tls_key)?;
 
+            let access_token = access_token.or_else(|| {
+                match abq_config::read_abq_config(get_abq_config_filepath()) {
+                    Some(abq_config) => Some(abq_config.rwx_access_token),
+                    None => None,
+                }
+            });
+
             let ResolvedConfig {
-                queue_addr: resolved_queue_addr,
                 token: resolved_token,
                 tls_cert: resolved_tls,
-            } = resolve_config(token, queue_addr, tls_cert, &access_token, &run_id)?;
+                rwx_access_token_kind,
+                queue_location,
+            } = resolve_config(token, queue_addr, tls_cert, tls_key, &access_token, &run_id)
+                .await?;
 
-            let client_auth = resolved_token.into();
+            if explicit_run_id_provided && !queue_location.is_remote() {
+                let mut cmd = Cli::command();
+                Err(cmd.error(
+                    ErrorKind::InvalidValue,
+                    "`abq test` was provided a run id, but we've detected an ephemeral queue.
+
+                    If you intended to run against a remote queue, please provide an access token by passing `--access-token`, setting `RWX_ACCESS_TOKEN`, or running `abq login`.
+                    If you intended to run against an ephemeral queue, please remove the run id argument.
```

Comment:
This might format a bit weird in the output - by default, rust treats leading whitespace in multiline output as significant.

One way to get around this is to use the `indoc` macro from the indoc library:

```suggestion
                    indoc!("
                    `abq test` was provided a run id, but we've detected an ephemeral queue.

                    If you intended to run against a remote queue, please provide an access token by passing `--access-token`, setting `RWX_ACCESS_TOKEN`, or running `abq login`.
                    If you intended to run against an ephemeral queue, please remove the run id argument.
                    ")
```

this will format with leading whitespace stripped.